core/api/tests/test_projects.py

```diff
@@ -129,6 +129,16 @@ def test_project_patch(self, user, project_url, project, agency):
         assert project.title == "Into the Spell"
         assert project.submission_category == "investment project"
         assert project.coop_agencies.count() == 1
+        assert project.generated_code == get_project_sub_code(
```

Comment:
this looks like it's in the end just testing `get_project_sub_code` against itself, but I guess with the addition that it's good to check that the generated_code was actually set properly.

Reply:
Yeah, this is what I wanted to test here, just to make sure that the code was regenerated after the project update.